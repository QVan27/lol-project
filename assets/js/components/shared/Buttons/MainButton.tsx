import React from "react";

const STATUS = {
  HOVERED: "hovered",
  NORMAL: "normal",
};

interface Props {
  page: string;
  children: string;
};

export default function MainButton(props: Props) {
  const [status, setStatus] = React.useState(STATUS.NORMAL);

  const onMouseEnter = () => {
    setStatus(STATUS.HOVERED);
  };

  const onMouseLeave = () => {
    setStatus(STATUS.NORMAL);
  };

  return (
    <a
      href={props.page}
      className={`button-lol ${status}`}
      onMouseEnter={onMouseEnter}
      onMouseLeave={onMouseLeave}
    >
      {props.children}
    </a>
  );
}
