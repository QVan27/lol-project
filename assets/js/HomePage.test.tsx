import * as React from "react";
import renderer from "react-test-renderer";
import MainButton from "../../assets/js/components/shared/Buttons/MainButton";

it("changes the class when hovered", () => {
  const component = renderer.create(
    <MainButton page="http://www.facebook.com">Facebook</MainButton>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  renderer.act(() => {
    tree.props.onMouseEnter();
  });
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  renderer.act(() => {
    tree.props.onMouseLeave();
  });
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
