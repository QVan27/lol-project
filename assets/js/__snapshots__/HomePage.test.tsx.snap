// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changes the class when hovered 1`] = `
<a
  className="button-lol normal"
  href="http://www.facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Facebook
</a>
`;

exports[`changes the class when hovered 2`] = `
<a
  className="button-lol hovered"
  href="http://www.facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Facebook
</a>
`;

exports[`changes the class when hovered 3`] = `
<a
  className="button-lol normal"
  href="http://www.facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Facebook
</a>
`;
